### PR TITLE
VIH-11065  preload conference cache with days hearings

### PR DESCRIPTION
**Pull request by will-craig**
### Jira link
https://tools.hmcts.net/jira/browse/VIH-11065

### Change description](https://tools.hmcts.net/jira/browse/VIH-11065)
Added a background service with daily/time delay, to orchestrate preloading the conference cache with all the days conference. Added tests and config settings

### CODE REVIEW

File: VideoWeb/VideoWeb.Common/VideoWeb.Common.csproj [hunk 1]
```diff
@@ -16,7 +16,7 @@
     </PackageReference>
     <PackageReference Include="StackExchange.Redis" Version="2.8.0" />
     <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="7.5.2" />
-    <PackageReference Include="VideoApi.Client" Version="3.0.3" />
+    <PackageReference Include="VideoApi.Client" Version="3.1.1-pr-0695-0003" />
```
Comment (marcogagliardi) [inline VideoWeb/VideoWeb.Common/VideoWeb.Common.csproj:19]: remember to change

File: VideoWeb/VideoWeb.Contract/VideoWeb.Contract.csproj [hunk 2]
```diff
@@ -13,7 +13,7 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
-    <PackageReference Include="VideoApi.Client" Version="3.0.3" />
+    <PackageReference Include="VideoApi.Client" Version="3.1.1-pr-0695-0003" />
```
Comment (marcogagliardi) [inline VideoWeb/VideoWeb.Contract/VideoWeb.Contract.csproj:16]: remember to change

File: VideoWeb/VideoWeb.UnitTests/VideoWeb.UnitTests.csproj [hunk 3]
```diff
@@ -22,7 +22,7 @@
     <PackageReference Include="NUnit" Version="4.1.0" />
     <PackageReference Include="NUnit3TestAdapter" Version="4.5.0" />
     <PackageReference Include="NBuilder" Version="6.1.0" />
-    <PackageReference Include="VideoApi.Client" Version="3.0.3" />
+    <PackageReference Include="VideoApi.Client" Version="3.1.1-pr-0695-0003" />
```
Comment (marcogagliardi) [inline VideoWeb/VideoWeb.UnitTests/VideoWeb.UnitTests.csproj:25]: remember to change

File: VideoWeb/VideoWeb.UnitTests/packages.lock.json [hunk 4]
```diff
@@ -2368,7 +2368,7 @@
           "Swashbuckle.AspNetCore": "[6.6.1, )",
           "Swashbuckle.AspNetCore.Annotations": "[6.6.1, )",
           "Swashbuckle.AspNetCore.Newtonsoft": "[6.6.1, )",
-          "VideoApi.Client": "[3.0.3, )",
+          "VideoApi.Client": "[3.1.1-pr-0695-0003, )",
```
Comment (marcogagliardi) [inline VideoWeb/VideoWeb.UnitTests/packages.lock.json:2371]: remember to change

File: VideoWeb/VideoWeb.UnitTests/packages.lock.json [hunk 5]
```diff
@@ -2387,14 +2387,14 @@
           "Microsoft.Identity.Client": "[4.61.0, )",
           "StackExchange.Redis": "[2.8.0, )",
           "System.IdentityModel.Tokens.Jwt": "[7.5.2, )",
-          "VideoApi.Client": "[3.0.3, )"
+          "VideoApi.Client": "[3.1.1-pr-0695-0003, )"
```
Comment (marcogagliardi) [inline VideoWeb/VideoWeb.UnitTests/packages.lock.json:2390]: remember to change

File: VideoWeb/VideoWeb.UnitTests/packages.lock.json [hunk 6]
```diff
@@ -2387,14 +2387,14 @@
...
         }
       },
       "videoweb.contract": {
         "type": "Project",
         "dependencies": {
           "BookingsApi.Client": "[3.0.5, )",
-          "VideoApi.Client": "[3.0.3, )",
+          "VideoApi.Client": "[3.1.1-pr-0695-0003, )",
```
Comment (marcogagliardi) [inline VideoWeb/VideoWeb.UnitTests/packages.lock.json:2397]: remember to change

File: VideoWeb/VideoWeb/VideoWeb.csproj [hunk 7]
```diff
@@ -48,7 +48,7 @@
     <PackageReference Include="Swashbuckle.AspNetCore" Version="6.6.1" />
     <PackageReference Include="Swashbuckle.AspNetCore.Annotations" Version="6.6.1" />
     <PackageReference Include="Swashbuckle.AspNetCore.Newtonsoft" Version="6.6.1" />
-    <PackageReference Include="VideoApi.Client" Version="3.0.3" />
+    <PackageReference Include="VideoApi.Client" Version="3.1.1-pr-0695-0003" />
```
Comment (marcogagliardi) [inline VideoWeb/VideoWeb/VideoWeb.csproj:51]: remember to change

File: VideoWeb/VideoWeb/packages.lock.json [hunk 8]
```diff
@@ -229,9 +229,9 @@
       },
       "VideoApi.Client": {
         "type": "Direct",
-        "requested": "[3.0.3, )",
-        "resolved": "3.0.3",
-        "contentHash": "VVlCmvcbyiMo5lP/KLNqi4T+VFe0CIfbhYr8YPm8WzKoXu97Cqp5hWDL2rLW/9PYn95UKvrDSC42BAPD8CjItQ==",
+        "requested": "[3.1.1-pr-0695-0003, )",
```
Comment (marcogagliardi) [inline VideoWeb/VideoWeb/packages.lock.json:232]: remember to change

File: charts/vh-video-web/values.dev.template.yaml [hunk 9]
```diff
@@ -10,4 +10,5 @@ java:
         EJUDAD__POSTLOGOUTREDIRECTURI: https://${SERVICE_FQDN}/logout
         EJUDAD__REDIRECTURI: https://${SERVICE_FQDN}/home
         DOM1__POSTLOGOUTREDIRECTURI: https://${SERVICE_FQDN}/logout
-        DOM1__REDIRECTURI: https://${SERVICE_FQDN}/home
\ No newline at end of file
+        DOM1__REDIRECTURI: https://${SERVICE_FQDN}/home
+        VHSERVICES__VIDEOAPIURL: https://vh-video-api-pr-695.dev.platform.hmcts.net/
```
Comment (marcogagliardi) [inline charts/vh-video-web/values.dev.template.yaml:14]: remember to change

**Review by marcogagliardi — APPROVED** [review]
LG2M

File: VideoWeb/VideoWeb.Common/ConferenceService.cs [hunk 10]
```diff
@@ -67,12 +68,24 @@ public async Task UpdateConferenceAsync(Conference conference, CancellationToken
     
     public async Task<IEnumerable<Conference>> GetConferences(IEnumerable<Guid> conferenceIds, CancellationToken cancellationToken = default)
     {
-        var ids = conferenceIds.ToArray();
-        return await Task.WhenAll(ids.Select(id => GetConference(id, cancellationToken)));
+        return await Task.WhenAll(conferenceIds.Select(id => GetConference(id, cancellationToken)));
```
[the remark this answers is not in the document]
Comment (will-craig) [reply VideoWeb/VideoWeb.Common/ConferenceService.cs:71]: I am, you're looking at the wrong function, PopulateConferenceCacheForToday on line 79

**Review by ronaldmalone — APPROVED** [review]

**Pull request merged**